lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -106,40 +105,76 @@ def clean(self):
         cleaned_data['user'] = self._clean_requested_user(cleaned_data, usage_key.course_key)
         return cleaned_data
 
+    def clean_username(self):
+        """
+        Return cleaned username.
+
+        Overrides the default behaviour that maps an empty string to None. This
+        allows us to differentiate between no username being provided (None) vs
+        an empty username being provided ('').
+        """
+        if 'username' in self.data:
```

Comment:
The reason for that is that we want to be able to distinguish between username = '' and username not being provided at all. The way Django forms code works is that if username is simply not provided as a query parameter, it will show up as an empty string in the cleaned_data, which is what we generally use. So we are unable to distinguish between the case where username is an empty string and when username is absent.

So what I"m doing here is:
- if username is absent, set username to None instead of an empty string
- If username is present use whatever the default string cleaning code returns.
If a